BLUESPAWN-client/headers/util/filesystem/FileSystem.h

```diff
@@ -12,13 +10,217 @@
 #include <fstream>
 #include <filesystem>
 #include <algorithm>
-
+#include <fileapi.h>
+#include <vector>
+#include <Shlwapi.h>
 #define BUFSIZE 1024
 #define MD5LEN  16
 
 using namespace std;
-namespace fs = std::experimental::filesystem::v1;
+//namespace fs = std::experimental::filesystem::v1;
+namespace FileSystem {
+	bool CheckFileExists(LPCWSTR);
+	
+	struct FileAttribs {
+		//long size;
+		//long permissions;
+		//bool hidden;
+		wstring extension;
+	};
+
+	struct FileSearchAttribs {
+		std::vector<wstring> extensions;
+	};
+
+	class File {
+		//Whether or not this current file actually exists
+		bool FileExists; 
+		//Path to the file
+		LPCWSTR FilePath;
+		//Handle for the file
+		HANDLE hFile;
```

Comment:
I recommend a HandleWrapper from wrappers.hpp. You use it just like a normal handle, but it automatically closes itself and manages reference counting.

What happens if I copy a File object and have two copies of the same file HANDLE and I destroy one? Does the other copy get invalidated?